### PR TITLE
feat: #3916 basic core of new config micro service

### PR DESCRIPTION
**Pull request by Shaun-Farrell**
Detail as per issue below (required):

#3916

* New core structure for config micro service

### CODE REVIEW

File: packages/config-service/package.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "config-service",
+  "version": "1.0.0",
+  "private": true,
+  "description": "Config Service",
+  "repository": "git@github.com:reapit/config-service.git",
+  "license": "MIT",
+  "author": "ltran@reapit.com",
```
Comment (willmcvay) [inline packages/config-service/package.json:8]: Author - you!

File: packages/config-service/package.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  "main": "index.js",
+  "scripts": {
+    "start": "yarn serverless offline start --runtime nodejs12.x --stage local --printOutput",
+    "test": "cross-env TZ=UTC jest --watch --colors --silent",
+    "build": "echo '...skipping...'",
+    "lint": "eslint --cache --ext=ts,tsx src --fix",
+    "release": "echo '...skipping...'",
```
Comment (willmcvay) [inline packages/config-service/package.json:15]: Should be serverless deploy

File: packages/config-service/serverless.yml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      Resource: 'arn:aws:ses:${self:provider.region}:${self:custom.env.AWS_ACCOUNT_ID}:parameter/Client*'
+
+resources:
+  Resources:
+    GatewayResponseDefault4XX:
+      Type: 'AWS::ApiGateway::GatewayResponse'
+      Properties:
+        ResponseParameters:
+          gatewayresponse.header.Access-Control-Allow-Origin: "'*'"
+          gatewayresponse.header.Access-Control-Allow-Headers: "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token,X-Amz-User-Agent,reapit-customer,api-version'"
```
Comment (willmcvay) [inline packages/config-service/serverless.yml:47]: don't think you need the custom headers here

File: packages/config-service/serverless.yml [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        ResponseType: DEFAULT_4XX
+        RestApiId:
+          Ref: 'ApiGatewayRestApi'
+    GatewayResponseDefault5XX:
+      Type: 'AWS::ApiGateway::GatewayResponse'
+      Properties:
+        ResponseParameters:
+          gatewayresponse.header.Access-Control-Allow-Origin: "'*'"
+          gatewayresponse.header.Access-Control-Allow-Headers: "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token,X-Amz-User-Agent,reapit-customer,api-version'"
```
Comment (willmcvay) [inline packages/config-service/serverless.yml:56]: as above

File: packages/config-service/serverless.yml [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+          cors:
+            origin: '*'
+            headers:
+              - Content-Type
+              - X-Amz-Date
+              - Authorization
+              - X-Api-Key
+              - X-Amz-Security-Token
+              - X-Amz-User-Agent
+              - reapit-customer
```
Comment (willmcvay) [inline packages/config-service/serverless.yml:93]: as above, don't need these

File: packages/config-service/src/controllers/__tests__/create-config.test.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+jest.mock('../../core/logger')
+
+const KEY = 'CLI/OpayoConfig'
+const mockConfigValue = JSON.stringify({ configTest: true })
+
+const baseMockReq = {
+  traceId: 'SOME_TRACE_ID',
+  headers: {
+    ['x-api-key']: 'SOME_API_KEY',
+    ['reapit-customer']: 'SOME_CODE',
```
Comment (willmcvay) [inline packages/config-service/src/controllers/__tests__/create-config.test.ts:20]: don't think we need this one

**Review by willmcvay — COMMENTED** [review]
Just a few bits of tidy up

File: packages/config-service/package.json (hunk 1 above)
Comment (Shaun-Farrell) [reply packages/config-service/package.json:8]: Done

File: packages/config-service/package.json (hunk 2 above)
Comment (Shaun-Farrell) [reply packages/config-service/package.json:15]: Done

File: packages/config-service/serverless.yml (hunk 3 above)
Comment (Shaun-Farrell) [reply packages/config-service/serverless.yml:47]: Removed reapit-customer,api-version

File: packages/config-service/src/controllers/__tests__/create-config.test.ts (hunk 6 above)
Comment (Shaun-Farrell) [reply packages/config-service/src/controllers/__tests__/create-config.test.ts:20]: Removed reapit-customer,api-version

File: packages/config-service/serverless.yml (hunk 4 above)
Comment (Shaun-Farrell) [reply packages/config-service/serverless.yml:56]: Removed reapit-customer,api-version

**Comment by Shaun-Farrell** [comment]
Going to close this PR and re-start with new changes.

**Pull request closed without merging**